backends-clickhouse/src/main/scala/org/apache/spark/sql/execution/datasources/v1/CHMergeTreeWriterInjects.scala

```diff
@@ -83,11 +86,14 @@ class CHMergeTreeWriterInjects extends GlutenFormatWriterInjectsBase {
       tableName,
       orderByKeyOption,
       lowCardKeyOption,
+      minmaxIndexKeyOption,
+      bfIndexKeyOption,
+      setIndexKeyOption,
       primaryKeyOption,
       partitionColumns,
       ConverterUtils.convertNamedStructJson(tableSchema),
       clickhouseTableConfigs,
-      dataSchema
+      tableSchema.toAttributes // use table schema instead of data schema
```

Comment:
@zzcclp , pay attention to this

Reply:
does it make sure that the `dataSchema` is the same as the tableSchema, especially the nullable property?
If yes, can remove the dataSchema parameter.

Reply:
yes, done